### PR TITLE
feat: implement `skipPaths` option for 'license' policy

### PR DESCRIPTION
**Pull request by smira**
This allows to skip directories which shouldn't be checked (e.g.
`.git/`) or directories which aren't readable.

While I was at it, bumped Go version, golangci-lint, refactored tests to
avoid duplicated code and make linter happy.

Rearranged a bit `Dockerfile` to avoid `go mod download` when `go.mod`
is not changed.

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

### CODE REVIEW

File: .conform.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ policies:
   spec:
     headerLength: 89
     dco: true
-    gpg: true
+    gpg: false
```
Comment (smira) [inline .conform.yaml]: switched it off to make `make enforce` pass on the repo itself

File: Makefile [hunk 2]
```diff
@@ -19,30 +19,34 @@ build:
 		-t conform/$@:$(TAG) \
 		--target=$@ \
 		$(COMMON_ARGS)
-	@docker run --rm -it -v $(PWD)/build:/build conform/$@:$(TAG) cp /conform-linux-amd64 /build
-	@docker run --rm -it -v $(PWD)/build:/build conform/$@:$(TAG) cp /conform-darwin-amd64 /build
```
Comment (smira) [inline Makefile]: `-it` shouldn't be there (breaks as there's no TTY in drone)

File: Makefile [hunk 3]
```diff
@@ -19,30 +19,34 @@ build:
...
 
 image: build
 	@docker build \
 		-t autonomy/conform:$(TAG) \
 		--target=$@ \
 		$(COMMON_ARGS)
 
+.PHONY: login
+login:
+	@docker login --username $(DOCKER_USERNAME) --password $(DOCKER_PASSWORD)
```
Comment (smira) [inline Makefile]: to hide docker creds

File: Makefile [hunk 4]
```diff
@@ -19,30 +19,34 @@ build:
...
+
 push: image
 	@docker tag autonomy/conform:$(TAG) autonomy/conform:latest
 	@docker push autonomy/conform:$(TAG)
 	@docker push autonomy/conform:latest
 
 deps:
-	@GO111MODULES=on CGO_ENABLED=0 go get -u github.com/autonomy/gitmeta
-	@GO111MODULES=on CGO_ENABLED=0 go get -u github.com/autonomy/conform
+	@GO111MODULE=on CGO_ENABLED=0 go get -u github.com/autonomy/gitmeta
```
Comment (smira) [inline Makefile]: typo in variable

File: internal/policy/commit/commit.go [hunk 5]
```diff
@@ -84,11 +84,9 @@ func (c *Commit) Compliance(options *policy.Options) (report policy.Report) {
 			return
 		}
 		msg = string(contents)
-	} else {
-		if msg, err = g.Message(); err != nil {
-			report.Errors = append(report.Errors, errors.Errorf("failed to get commit message: %v", err))
-			return
-		}
+	} else if msg, err = g.Message(); err != nil {
```
Comment (smira) [inline internal/policy/commit/commit.go]: linter thinks it's better, I don't quite agree...

Comment (andrewrynhard) [reply internal/policy/commit/commit.go]: +1

**Pull request merged**